### PR TITLE
chore(deps): bump go from 1.18 to 1.20

### PR DESCRIPTION
**Pull request by lahabana**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: mk/build.mk [hunk 1]
```diff
@@ -11,7 +11,7 @@ build_info_fields := \
 build_info_ld_flags := $(foreach entry,$(build_info_fields), -X github.com/kumahq/kuma/pkg/version.$(entry))
 
 LD_FLAGS := -ldflags="-s -w $(build_info_ld_flags) $(EXTRA_LD_FLAGS)"
-CGO_ENABLED := 0
+EXTRA_GOENV=GOEXPERIMENT=boringcrypto
```
Comment (lahabana) [inline mk/build.mk:14]: will need to remove this

**Review by slonka — COMMENTED** [review]
The CI is all red so I'm guessing this needs a bit more work.

**Comment by lahabana** [comment]
Yeah the CP fails to start for a weird reason :(

**Comment by slonka** [comment]
> Yeah the CP fails to start for a weird reason :(

logs day 

> 2023-03-13T08:18:12Z logger.go:66: exec /usr/bin/kuma-cp: no such file or directory

**Comment by slonka** [comment]
I think binary is broken because we compile it and then copy to docker. We still need to explicitly set CGO_ENABLED=0, this commit makes it work https://github.com/kumahq/kuma/pull/6256/commits/2c8021527904620ecf48cd2dd33cae9ef6348408. Feel free to cherry pick it.

**Comment by lahabana** [comment]
Ha right. Some future reading: https://github.com/golang/go/issues/28065
Also it was a surprise to me to discover that `CGO_ENABLED=1` by default

**Comment by slonka** [comment]
Just FYI, I need go >= 1.19 to try out pgx, that's why I picked this up.

**Comment by slonka** [comment]
Format is not working, I'm going to push it myself.

File: mk/generate.mk [hunk 2]
```diff
@@ -128,4 +128,4 @@ generate/test-server:
 # _DELETE_GO_EMBED_WORKAROUND_ part after generation
 .PHONY: generate/fix-embed
 generate/fix-embed:
-	find $(POLICIES_DIR) -name zz_generated.resource.go -type f -exec $(SHELL) -c "sed -i.bak 's/_DELETE_GO_EMBED_WORKAROUND_//g' {} && rm {}.bak" \;
+	find $(POLICIES_DIR) -name zz_generated.resource.go -type f -exec $(SHELL) -c "sed -i.bak 's/ _DELETE_GO_EMBED_WORKAROUND_//g' {} && rm {}.bak" \;
```
Comment (slonka) [inline mk/generate.mk:131]: nitpick: why do we need this change?

**Review by slonka — APPROVED** [review]

File: mk/build.mk [hunk 3]
```diff
@@ -11,7 +11,9 @@ build_info_fields := \
 build_info_ld_flags := $(foreach entry,$(build_info_fields), -X github.com/kumahq/kuma/pkg/version.$(entry))
 
 LD_FLAGS := -ldflags="-s -w $(build_info_ld_flags) $(EXTRA_LD_FLAGS)"
-CGO_ENABLED := 0
+#EXTRA_GOENV=GOEXPERIMENT=boringcrypto
```
Comment (lahabana) [inline mk/build.mk:14]:
```suggestion
```

**Pull request merged**